inference/services/KernelFactory.cs

```diff
@@ -39,35 +50,40 @@ private Kernel CreateKernel(HttpClient httpClient)
         return kernalBuilder.Build();
     }
 
-    public async Task<Kernel> GetOrCreateKernelForInferenceAsync(CancellationToken cancellationToken = default)
+    public async Task<Kernel> GetOrCreateKernelForInferenceAsync(CancellationToken cancellationToken = default, int? index = null)
```

Comment:
Cancellation Token is always supposed to be last.

Reply:
updated with cancellation token as last par.